src/lua/fio.lua

```diff
@@ -200,7 +200,7 @@ fio.open = function(path, flags, mode)
         flags = { flags }
     end
     if type(mode) ~= 'table' then
-        mode = { mode or (bit.band(0x1FF, fio.umask())) }
+        mode = { mode or tonumber('666', 8) }
```

Comment:
Actually I don't know what to add here, as the previous code was just wrong. Mainly, I spent the time proofing that.

Reply:
I looked briefly around the changes around fio.open() and found the following:

#1211 -- it was fixed in 959ead30d6fa453aa7d48f7acb1e454f8df1657d (with mode 0777 instead of 0666, but anyway). Than regressed in 583e8ba2b3e0d3e919c136b6c167e41196fb5751. fio.mkdir also was fixed (d92e61bb485d3c768e78e351c78d7b30117b67fa), but everything is fine there now (no regression). There is also related 3e29e935ab83aa9888f6b8979d2e7deacf6bfbbc.

My thought was that you looked over the same issues/commits and at this point it is cheap to sum up this part of the history. If your way was different, my guess is wrong and I dismiss my comment.

Reply:
> Than regressed in https://github.com/tarantool/tarantool/commit/583e8ba2b3e0d3e919c136b6c167e41196fb5751.

https://github.com/tarantool/tarantool/commit/583e8ba2b3e0d3e919c136b6c167e41196fb5751 closed https://github.com/tarantool/tarantool/issues/2757 where this change is stated:

> fio.open() default mode is 0, which is non-workable. If mode is nil, like in open() the default mode should be used - 777 & umask

Reply:
We usually create temporary file or export data. Creating an executable file is a rare case, so it is OK to ask for it explicitly. The quotes proposal from #2757 is not the best option IMHO.

Reply:
Well, some 'we know about previous proposals (and solutions, and regressions), but decided differently here' is desirable in the change description, but I don't want to spin around it, so it is up to the developer.